### PR TITLE
Func

### PR DESCRIPTION
**Pull request by maulakhan**
Concept map is still not working for me

### CODE REVIEW

**Review by tischi — COMMENTED** [review]
I think we should also add another more exercise where there is some code and the students need to replace the repetitive part by a function.

File: _includes/script_functions/activities/Function_display_slice.ijm [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	
+
+// a function that calculates mean intensity of a specified slice in a stack and displays it
+function calculateSliceMeanIntensity(sliceNumber)
+{
+	setSlice(sliceNumber);
+	run("Duplicate...", "title=slice"+sliceNumber);
+	sliceName = "slice"+sliceNumber;
+	selectWindow(sliceName);
+	run("8-bit");
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _includes/script_functions/activities/Function_display_slice.ijm]: removed 8-bit command

File: _includes/script_functions/activities/Function_display_slice.ijm [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	getStatistics(area, mean);
+	return mean;
+
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _includes/script_functions/activities/Function_display_slice.ijm]: removed

File: _includes/script_functions/activities/Function_display_slice.ijm [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+run("MRI Stack"); // load the example MRI image stack
+imageName = getTitle(); //get the image title 
+
+//select first slice,calculate its mean intensity and display it
+selectWindow(imageName);
+sliceNumber = 1;	
+meanIntensitySlice1 = calculateSliceMeanIntensity(sliceNumber);
+print("Slice#"+ sliceNumber + " has mean intensity: " + meanIntensitySlice1);
+
+//select last slice,calculate its mean intensity and display it
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _includes/script_functions/activities/Function_display_slice.ijm]: space added

File: _modules/script_functions.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+---
+title:     Script functions
+layout:    module
+prerequisites:
+  - "[Basic properties of images and pixels](../pixels)"
+
+objectives:
+  - "Describe the purpose of a function in a script."
+  - "Demonstrate the usage and flow of a function."
+  - "Make the script more efficient."
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _modules/script_functions.md:10]: changed

File: _modules/script_functions.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+---
+title:     Script functions
+layout:    module
+prerequisites:
+  - "[Basic properties of images and pixels](../pixels)"
+
+objectives:
+  - "Describe the purpose of a function in a script."
+  - "Demonstrate the usage and flow of a function."
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _modules/script_functions.md]: yes, thats much better, added already

File: _modules/script_functions.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  - "Make the script more efficient."
+motivation: |
+  When a series of steps(i.e. algorithm) that has to be performed on an image or a set of images should be executed more than once, or when the script gets too long and repetitive, it is more efficient to write this series as modules called "functions".
+  This essentially means that you can reuse parts of code instead of rewriting it. A function is a block that can be called with inputs and can return values.
+
+concept_map: >
+  graph TD
+    IV("Input values-None") --> F(Function (Set of operations))
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _modules/script_functions.md]: Yes this solves the problem. Done!

File: _modules/script_functions.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    F --> OV("Output values-None")
+
+figure: /figures/Function_demonstration.png
+figure_legend: Components and working of a function within a script.
+
+activity_preface: |
+  - Run the script and try to calculate mean intensity of 2 different slices.
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _modules/script_functions.md]: Added more text to elaborate!

File: _modules/script_functions.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+activities:
+  - ["ImageJ Macro", "script_functions/activities/Function_display_slice.ijm", "java"]
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _modules/script_functions.md]: Done!

File: _modules/script_functions.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  - ["ImageJ Macro", "script_functions/exercises/functions_imagejmacro.md"]
+
+assessment: >
+
+  ### True or False
+    - The values returned by function cannot be used outside of the function.
+    - Function can be called multiple times within a script.
+
+    > ## Solution
+    >   - It can be used unless the variable returned is not overwritten. **False**
```
[the remark this answers is not in the document]
Comment (maulakhan) [reply _modules/script_functions.md]: This one was confusing so I changed to another question.

**Comment by maulakhan** [comment]
@tischi Can you please take a look at the new commits.

**Comment by tischi** [comment]
Thanks @maulakhan !

**Pull request merged**